core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -1037,71 +1037,75 @@ public void testCatalogTokenRefresh() throws Exception {
     catalog.initialize(
         "prod", ImmutableMap.of(CatalogProperties.URI, "ignored", "credential", "catalog:secret"));
 
-    Thread.sleep(3_000); // sleep until after 2 refresh calls
-
-    // call client credentials with no initial auth
-    Mockito.verify(adapter)
-        .execute(
-            eq(HTTPMethod.POST),
-            eq("v1/oauth/tokens"),
-            any(),
-            any(),
-            eq(OAuthTokenResponse.class),
-            eq(emptyHeaders),
-            any());
-
-    // use the client credential token for config
-    Mockito.verify(adapter)
-        .execute(
-            eq(HTTPMethod.GET),
-            eq("v1/config"),
-            any(),
-            any(),
-            eq(ConfigResponse.class),
-            eq(catalogHeaders),
-            any());
-
-    // verify the first token exchange
-    Map<String, String> firstRefreshRequest =
-        ImmutableMap.of(
-            "grant_type", "urn:ietf:params:oauth:grant-type:token-exchange",
-            "subject_token", "client-credentials-token:sub=catalog",
-            "subject_token_type", "urn:ietf:params:oauth:token-type:access_token",
-            "scope", "catalog");
-    Mockito.verify(adapter)
-        .execute(
-            eq(HTTPMethod.POST),
-            eq("v1/oauth/tokens"),
-            any(),
-            Mockito.argThat(firstRefreshRequest::equals),
-            eq(OAuthTokenResponse.class),
-            eq(catalogHeaders),
-            any());
-
-    // verify that a second exchange occurs
-    Map<String, String> secondRefreshRequest =
-        ImmutableMap.of(
-            "grant_type", "urn:ietf:params:oauth:grant-type:token-exchange",
-            "subject_token", "token-exchange-token:sub=client-credentials-token:sub=catalog",
-            "subject_token_type", "urn:ietf:params:oauth:token-type:access_token",
-            "scope", "catalog");
-    Map<String, String> secondRefreshHeaders =
-        ImmutableMap.of(
-            "Authorization",
-            "Bearer token-exchange-token:sub=client-credentials-token:sub=catalog");
-    Mockito.verify(adapter)
-        .execute(
-            eq(HTTPMethod.POST),
-            eq("v1/oauth/tokens"),
-            any(),
-            Mockito.argThat(secondRefreshRequest::equals),
-            eq(OAuthTokenResponse.class),
-            eq(secondRefreshHeaders),
-            any());
+    Awaitility.await()
```

Comment:
the default poll interval is to poll every 200 millis with an initial delay of 100 millis. I think this is fine for these tests here but not that this can be modified as outlined in https://github.com/awaitility/awaitility/wiki/Usage#polling